sensor/common/registry/registry_store.go

```diff
@@ -68,15 +72,8 @@ func (rs *Store) getRegistries(namespace string) registries.Set {
 	return regs
 }
 
-// UpsertRegistry upserts the given registry with the given credentials in the given namespace into the store.
-func (rs *Store) UpsertRegistry(ctx context.Context, namespace, registry string, dce config.DockerConfigEntry) error {
-	secure, err := rs.checkTLS(ctx, registry)
-	if err != nil {
-		return errors.Wrapf(err, "unable to check TLS for registry %q", registry)
-	}
-
-	regs := rs.getRegistries(namespace)
-	err = regs.UpdateImageIntegration(&storage.ImageIntegration{
+func createImageIntegration(registry string, dce config.DockerConfigEntry, secure bool) *storage.ImageIntegration {
```

Comment:
I think you've mentioned making all these as Docker integrations does not always works (ex: ECR integrations need periodic token updates). Can we add a comment talking about this?

Reply:
I haven't proven that yet, thats a theory at this point - if I do prove it I'll update the comment accordingly.